single-server/single-server.md

```diff
@@ -0,0 +1,68 @@
+# Single Server Study
+
+This is a POC study following the [Single Server Deployment Study](https://docs.google.com/document/d/1Fgiz-TRBn4L4Ur00j-v7TcloL6v6L86Xc4IbFr3Qde4/edit?ts=5e86e740#heading=h.adzuc1cjggcf) made in April, 2020. Purpose of this study is to find out what are the steps to restore cluster status after OS upgrade in single server deployment.
+
+K8s cluster for the testing is generated using ```kubeadm``` tool and does not include any **CAPM3**, **CAPI** or **METAL3-DEV-ENV** related components or workflows.
+
+### Backup the data
+
+To be able to initialize a new cluster after upgrading with the old state, we need to backup following data items.  
+* The root certificate files ```/etc/kubernetes/pki/ca.crt``` and ```/etc/kubernetes/pki/ca.key```.
+* The etcd data.
+
+We enable etcd backup by mounting extra disk to etcd's default [--data-dir](https://etcd.io/docs/v3.4/op-guide/configuration/#--data-dir) ```/var/lib/etcd``` before running ```kubeadm init```. Backing up the root certificate is a one-time operation that is done manually after creating the master with kubeadm init. We copied ```/etc/kubernetes/pki/ca.crt``` and ```/etc/kubernetes/pki/ca.key``` to extra disks.
+
+### Manual steps to preserve cluster data
+
+1. Create VM with two bootable disks. **Note:** We actually used the same disk image to create these disks. 
+
+2. Add new "data" disk to VM. We took following steps after disk creation: 
+
+* ```sudo (echo n; echo p; echo 1; echo  ;echo  ;echo w) | fdisk /dev/<device_name>```
+* ```sudo mkfs.ext4 /dev/<device_name>``` 
+* ```mkdir var/lib/etcd```
+* ```sudo mount /dev/<device_name> /var/lib/etcd```
+
+3. Install kubeadm and set up the environment: 
+* These instructions was used to install ```kubeadm``` [Installing kubeadm](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/)
+* Set /proc/sys/net/ipv4/ip_forward content to 1
+	
+4. Run kubeadm init ```sudo kubeadm init --ignore-preflight-errors=DirAvailable--var-lib-etcd```. Always use ```--ignore-preflight-errors=DirAvailable--var-lib-etcd``` flag with kubeadm init. Otherwise initialization will fail, if previously mounted disk's file system has any content. For example in case of ext4 ```lost+found``` folder is created during ```sudo mkfs.ext4 /dev/<device_name>```. 
```

Comment:
Shoud it be `sudo kubeadm init --ignore-preflight-errors=DirAvailable --var-lib-etcd` (there should be a space between DirAvailable and --var-lib-etcd)?